pkg/orchestrator/assetscanwatcher/watcher.go

```diff
@@ -367,21 +373,32 @@ func (w *Watcher) cleanupResources(ctx context.Context, assetScan *models.AssetS
 		var retryableError provider.RetryableError
 		switch {
 		case errors.As(err, &fatalError):
-			assetScan.ResourceCleanup = utils.PointerTo(models.ResourceCleanupStateFailed)
+			assetScan.ResourceCleanupStatus = models.NewResourceCleanupStatus(
+				models.ResourceCleanupStatusStateFailed,
+				"Resource cleanup failed.",
+				fatalError.Error(),
```

Comment:
By machine readable we mean something that was generated by an error and not handwritten, and here this happens.

Reply:
By "machine readable" we mean something that is machine parsable as a simple string like the "reason" field from Kubernetes conditions. https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions

For example it could be something like "UnexpectedError" or "ResourceCleanupFailed" or "OneOrMoreCleanupStepsFailed".

These are machine parsable fixed strings so that it doesn't matter what human readable text is in the "message" field. The client code reading this field can do:
```
if err.Reason == "UnexpectedError" {
 ...
}
```